### ISSUE TITLE
Cache-size database connection oracle with error! Help-me!

### ISSUE THREAD

**Issue by santiagoferraz** [yajra/laravel-oci8#443]
### Problem or Feature Request Summary

I want the primary key to be always incremented by 1 in ascending order for new inserts in the database.

My database has cache-size of 20, and with this setting when doing insert the primary key does not faithfully obey a growing order.

For what I researched, cache-size is responsible for perfomance, so I would not like to remove it.

It seems to me that the laravel application, rather than picking up the first ID in ascending order, takes any random ID at the time of generating Sequence (DB :: getSequence).

Help me find a solution please, I've already been 3 days with this problem!

### System Details

- Operating System: Ubuntu
- PHP version: 7.1
- Laravel Version: 5.4
- Laravel-OCI8 Version: 5.4

**Comment by yajra** [comment]
> It seems to me that the laravel application, rather than picking up the first ID in ascending order, takes any random ID at the time of generating Sequence (DB :: getSequence).

I think there is something wrong with your assessment (but I maybe wrong too). AFAIK, sequence will always increment to the next step value (default is step 1) every time you issue a `nextVal` sql. If you want it to increment properly, you need to make sure that all `insert` action you issue are correctly executed. **Note** that any failed insert execution will yield to an increment in the sequence current value which maybe the reason you think it's getting a random value?

**Comment by santiagoferraz** [comment]
The form of insertion is correct. When I give nextValue through the application it works correctly, incrementing one by one.

When the application is accessed the correct one is to cache a database of 20 ID's and each user session make a get into an ID using the nextValue. But it seems to me that the application is doing the cache, if there are 2 users the cache gets 20 for each, 40 of them in total.

I tried in many ways and could not use getSequence. I made a direct ID insert, where ID = select max (id) + 1.
